### PR TITLE
Fixes people going bald every round

### LINKED ISSUE #6463: Character setup occasionally makes you bald without warning
Been having to check character setup before joining or observing for so long that I'd forgotten this was an issue, not a feature.

This issue probably already exists somewhere deep within the pile of open issues, but I think it's worth reposting again, especially due to feature freeze.

### LINKED ISSUE #9358: Bald is more bald than ever before
Character setup almost always removes your hair, each and every round. As a result, the entire station is usually bald. I know that this has always been sort of an issue, but it's never been THIS frequent.

### PR DESCRIPTION
**Pull request by Intigracy**
Species datums weren't being generated before preferences to be able to reference the species allowed hair, so it was making everyone bald.

This makes it so you can't click on the three buttons you get when joining until your preferences are loaded, and your preferences actually load correctly now.

fixes #6463
fixes #9358

### CODE REVIEW

**Comment by ihadtoregisterforthis** [comment]
The endless cycle of faster ready-up/slower ready-up

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -182,14 +182,30 @@ var/const/MAX_SAVE_SLOTS = 8
...
-					return
+			if(canloadprefs)
+				var/load_pref = load_preferences_sqlite(C.ckey)
+				if(load_pref)
+					if(load_save_sqlite(C.ckey, C, default_slot))
+						caneditprefs = 1
+						to_chat(C,"<span class='notice'><b>You may now setup your character and select ready.</b></span>")
+						return
+			else
+				spawn(100)
```
Comment (clusterfack) [inline code/modules/client/preferences.dm]: Just do a while(!canloadprefs) sleep(10)

**Comment by Intigracy** [comment]
Oh I broke something.

Cry.

**Comment by Intigracy** [comment]
Took forever but I fixed it again.

Now with just one global variable.

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -182,14 +182,23 @@ var/const/MAX_SAVE_SLOTS = 8
...
-		if(!IsGuestKey(C.key))
-			var/load_pref = load_preferences_sqlite(C.ckey)
-			if(load_pref)
-				if(load_save_sqlite(C.ckey, C, default_slot))
-					return
-		randomize_appearance_for()
-		real_name = random_name(gender)
-		save_character_sqlite(src, C.ckey, default_slot)
+		if(!speciesinit)
+			spawn(10)
```
Comment (clusterfack) [inline code/modules/client/preferences.dm]: Why do you have nested spawns

**Comment by Intigracy** [comment]
fixed nested spawn

File: code/modules/client/preferences.dm (hunk 2 above)
Comment (clusterfack) [inline code/modules/client/preferences.dm]: Either way this should be a while loop and not a recursive function, there's absolutely no reason for it to be a recursive function when you can just delay it with a while loop

Comment (Intigracy) [inline code/modules/client/preferences.dm]:
```
        while(!speciesinit)
            spawn(10)
```

code\modules\client\preferences.dm:187:error: : invalid expression

```
        while(!speciesinit)
            sleep(10)
```

can't join the game until speciesinit returns true

**Pull request merged**